tests/unit/parts/test_yaml_utils.py

```diff
@@ -255,3 +255,16 @@ def test_get_snap_project(project, project_dir, new_dir):
     actual_project = yaml_utils.get_snap_project(project_dir)
 
     assert actual_project == project
+
+
+@pytest.mark.parametrize("project", yaml_utils._SNAP_PROJECT_FILES)
+@pytest.mark.parametrize("project_dir", [None, "test-project-dir"])
+def test_get_snap_project_snap_not_a_directory(project, project_dir, new_dir):
+    project_dir = pathlib.Path(project_dir) if project_dir else new_dir
+    project_dir.mkdir(parents=True, exist_ok=True)
+    (project_dir / "snap").touch()
+
+    with pytest.raises(Exception) as raised:
```

Comment:
Let's use a specific exception so we know it will get caught correctly. You'll have to import this at the top of the file.

```suggestion
    with pytest.raises(craft_application.errors.ProjectDirectoryTypeError) as raised:
```